crates/sui-indexer/src/framework/runner.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use sui_rest_api::CheckpointData;
+
+use super::interface::Handler;
+
+// pub struct IndexerRunner {
+//     client: Client,
+//     handlers: Vec<Box<dyn Handler>>,
+// }
+
+// impl IndexerRunner {
+//     pub fn new(client: Client) -> Self {
+//         Self {
+//             client,
+//             handlers: Vec::new(),
+//         }
+//     }
+
+//     pub fn add_handler<T: Handler + 'static>(&mut self, handler: T) {
+//         self.handlers.push(Box::new(handler));
+//     }
+
+//     pub async fn run(self) -> Result<()> {
+//         todo!()
+//     }
+// }
+
+pub async fn run<S>(mut stream: S, mut handlers: Vec<Box<dyn Handler>>)
+where
+    S: futures::Stream<Item = CheckpointData> + std::marker::Unpin,
+{
+    use futures::StreamExt;
+
+    while let Some(checkpoint) = stream.next().await {
+        //TODO run handlers concurently
+        //TODO create tracing spans for processing
+        for handler in &mut handlers {
+            handler.process_checkpoint(&checkpoint).await.unwrap();
+        }
```

Comment:
to add on that, do we want to have configurable checkpoint data, like skipping object contents for example?

Reply:
> Can we have one runner for each handler so they don't block each other?

So this may require a bit of discussion. I agree that each handler should be able to be run concurrently (and that should be easy to fix here). One thing to note, however, is that due to the work you did a week or so ago the commit phase is decoupled from these "handlers" via channels in both cases. The question is, is that sufficient to absorb any potential slow downs? If not, what is the expected behavior for fetching data given we'd probably like to avoid fetching the checkpoint data more than a single time?

> to add on that, do we want to have configurable checkpoint data, like skipping object contents for example?

I don't think thats a necessary optimization we need to make right now as it would then complicate the Handler api (and the huge benefit of even having the Handler api is consistency). That and the new interface between indexer and FN should be much more efficient, reducing the overhead of fetching the object data.